packages/frontend/src/components/staking/StakingContainer.js

```diff
@@ -203,6 +205,14 @@ export function StakingContainer({ history, match }) {
         dispatch(handleUpdateCurrent(accountId));
     };
 
+    useEffect(() => {
+        if (!accountId || !validators.length) return;
+        
+        validators
+            .filter((validator) => validator.version === FARMING_VALIDATOR_VERSION && validator.active)
+            .forEach((validator) => dispatch(getValidatorFarmData(validator, accountId)));
+    }, [accountId, validators]);
```

Comment:
I prefer this solution for now. It is not very beautiful, I don't like it this way. But batching all actions in one dispatch may cause problems in the future. Now we can inspect each fetch action and we don't really care if some of them fail for some reason (as it fails now for incorrect validators). I thought a lot about this part of the code. And finally decided to let it as it is. 
It might be that perfect is the enemy of the good in this case.

Reply:
What I mean is to create a thunk that will handle dispatching all actions one by one like it is now. Thunk will:
 - select validators and filter validators using a selector
 - select `accountId` using selector
 - then it will dispatch `getValidatorFarmData` actions one by one like it is now.
It's just that all of it will happen in a separate thunk instead of the component hook.

Generally, we are trying to extract this type of functionality into redux thunks.

Also, we don't need to check if `!validators.length` is true because if so the forEach will do nothing.

Reply:
But that's not a blocker. Just a suggestion.

Reply:
@askulikov The `validator.active` condition here should be removed. This is currently not accumulating unclaimed rewards from inactive validators due to this. To repro you can confirm that ILLU rewards from domanodes validator are currently not added in askulikov.testnet